src/tests/jsx.rs

```diff
@@ -186,6 +186,76 @@ to!(
     "#
 );
 
+to!(
+    jsx_ph_labels,
+     r#"
+       import { Trans, ph } from "@lingui/react/macro";
+
+       <Trans>Refresh {ph({foo})} inbox</Trans>;
+       <Trans>Refresh {ph({foo: foo.bar})} inbox</Trans>;
+       <Trans>Refresh {ph({foo: expr()})} inbox</Trans>;
+       <Trans>Refresh {ph({foo: bar, baz: qux})} inbox</Trans>;
+       <Trans>Refresh {ph({})} inbox</Trans>;
+       <Trans>Refresh {ph({...spread})} inbox</Trans>;
+     "#,
+    r#"
+       import { Trans as Trans_ } from "@lingui/react";
+       <Trans_ message={"Refresh {foo} inbox"} id={"rtxU8c"}
+        values={{
+          foo: foo,
+        }}/>;
+       <Trans_ message={"Refresh {foo} inbox"} id={"rtxU8c"}
+        values={{
+          foo: foo.bar,
+        }}/>;
+       <Trans_ message={"Refresh {foo} inbox"} id={"rtxU8c"}
+        values={{
+          foo: expr(),
+        }}/>;
+       <Trans_ message={"Refresh {foo} inbox"} id={"rtxU8c"}
+        values={{
+          foo: bar,
+        }}/>;
+       <Trans_ message={"Refresh {0} inbox"} id={"AmeQ8b"}
+        values={{
+          0: {},
+        }}/>;
+       <Trans_ message={"Refresh {0} inbox"} id={"AmeQ8b"}
+        values={{
+          0: {
+            ...spread,
+          },
+        }}/>;
+    "#
+);
+
+to!(
+    jsx_nested_labels,
+     r#"
+       import { Trans, ph } from "@lingui/react/macro";
+
+       <Trans>Refresh <span>{{foo}}</span> inbox</Trans>;
```

Comment:
it seems like this is also working for some reason? the PR for js implementation mentioned that it won't work 😅

Reply:
It wont work because of typescript will not allow you to write this. ReactNode could not be an arbitrary object `{}`

Reply:
ah I see, Trans has special type for children https://github.com/lingui/js-lingui/blob/d36c323c2151f91b05305809bc9a6d291528aed2/packages/react/macro/index.d.ts#L14